### PR TITLE
enhancement #672: Changed user contacts endpoints logic

### LINKED ISSUE #672: Change user contacts endpoints logic
Change endpoints  /v2/users/{userId}/contacts/{name}, so they will work with contact id instead contact name

### PR DESCRIPTION
**Pull request by PavloVlasiuk**
- changed /v2/users/{userId}/contacts/{name} patch endpoint to /v2/users/{userId}/contacts/{contactId}
- changed /v2/users/{userId}/contacts/{name} delete endpoint to /v2/users/{userId}/contacts/{contactId}

closes #672

### CODE REVIEW

File: src/v2/api/pipes/ContactByUserIdPipe.ts [hunk 1]
```diff
@@ -12,17 +12,17 @@ export class ContactByUserIdPipe implements PipeTransform {
...
     if (!user) {
       throw new InvalidEntityIdException('user');
     }
 
-    const contactName = this.contactRepository.getContact(userId, name);
-    if (!contactName) {
-      throw new InvalidContactNameException();
+    const contact = this.contactRepository.getContact(userId, contactId);
+    if (!contact) {
+      throw new InvalidContactIdException();
```
Comment (JessFreak) [inline src/v2/api/pipes/ContactByUserIdPipe.ts:25]: describe this exception in swagger

File: src/v2/api/controllers/UserController.ts [hunk 2]
```diff
@@ -197,20 +197,20 @@ export class UserController {
   }
```
Comment (JessFreak) [inline src/v2/api/controllers/UserController.ts:198]: add swagger documantatin

File: src/v2/api/controllers/UserController.ts [hunk 3]
```diff
@@ -197,20 +197,20 @@ export class UserController {
...
+  @Patch('/:userId/contacts/:contactId')
   updateContact (
     @Param(ContactByUserIdPipe) params,
     @Body() body: UpdateContactDTO,
   ) {
-    return this.userService.updateContact(params.userId, params.name, body);
+    return this.userService.updateContact(params.userId, params.contactId, body);
   }
 
   @Access(PERMISSION.USERS_$USERID_CONTACTS_DELETE)
```
Comment (JessFreak) [inline src/v2/api/controllers/UserController.ts:208]: add swagger documantatin

**Review by JessFreak — CHANGES_REQUESTED** [review]

File: src/v2/api/controllers/UserController.ts [hunk 4]
```diff
@@ -197,20 +197,20 @@ export class UserController {
   }
 
   @Access(PERMISSION.USERS_$USERID_CONTACTS_UPDATE)
-  @Patch('/:userId/contacts/:name')
+  @Patch('/:userId/contacts/:contactId')
```
Comment (JessFreak) [inline src/v2/api/controllers/UserController.ts:200]: add swagger documentation

File: src/v2/api/controllers/UserController.ts [hunk 5]
```diff
@@ -197,20 +197,20 @@ export class UserController {
...
     @Param(ContactByUserIdPipe) params,
     @Body() body: UpdateContactDTO,
   ) {
-    return this.userService.updateContact(params.userId, params.name, body);
+    return this.userService.updateContact(params.userId, params.contactId, body);
   }
 
   @Access(PERMISSION.USERS_$USERID_CONTACTS_DELETE)
-  @Delete('/:userId/contacts/:name')
+  @Delete('/:userId/contacts/:contactId')
```
Comment (JessFreak) [inline src/v2/api/controllers/UserController.ts:209]: add swagger documentation

**Review by JessFreak — CHANGES_REQUESTED** [review]

**Review by JessFreak — COMMENTED** [review]
image
fill this cells

**Review by JessFreak — APPROVED** [review]

File: src/v2/api/controllers/TelegramGroupController.ts [hunk 6]
```diff
@@ -152,8 +148,7 @@ export class TelegramGroupController {
       Group with such id is not found
       
     DataNotFoundException:
-      Data were not found
```
Comment (Yakorrr) [inline src/v2/api/controllers/TelegramGroupController.ts:155]: data WAS not found :)
but it's unnecessary to change

**Review by Yakorrr — APPROVED** [review]

**Review by fokaaas — APPROVED** [review]

**Pull request merged**